src/include/detail/linalg/tdb_matrix.h

```diff
@@ -114,6 +114,26 @@ class tdbBlockedMatrix : public MatrixBase {
   // size_t pending_row_offset{0};
   // size_t pending_col_offset{0};
 
+  static size_t get_last_row(
```

Comment:
Hmm.  I don't think we want to open and close the array to get this info, only to open the array again.

However, I think this is a moot point, per the extensive comment above, as I think we should move everything into one constructor and use `std::optional` to indicate whether or not we want begin/end of the array.

Reply:
Sure, updated to new approach.